Sources/MessagingInApp/Gist/Managers/MessageManager.swift

```diff
@@ -57,271 +74,284 @@ class MessageManager: EngineWebDelegate {
-                            // This is only needed when message is dismissed and not when it is reset.
-                            if case .dismissed = messageState {
-                                self.gist.fetchUserMessagesFromRemoteQueue()
-                            }
-                        }
+                        // Dismiss the message from subclass
+                        self.onMessageDismissed(messageState: messageState)
                     }
-
                 default:
                     break
                 }
             }
-            inAppMessageManager.subscribe(keyPath: \.currentMessageState, subscriber: subscriber)
+            self.inAppMessageManager.subscribe(keyPath: \.currentMessageState, subscriber: subscriber)
             return subscriber
         }()
     }
 
-    /// Unsubscribes from InAppMessageState so that MessageManager can be deallocated.
-    func unsubscribeFromInAppMessageState() {
+    open func unsubscribeFromInAppMessageState() {
         guard let subscriber = inAppMessageStoreSubscriber else { return }
-
-        logger.logWithModuleTag("Unsubscribing MessageManager from InAppMessageState", level: .debug)
+        logger.logWithModuleTag("Unsubscribing BaseMessageManager from InAppMessageState", level: .debug)
         inAppMessageManager.unsubscribe(subscriber: subscriber)
         inAppMessageStoreSubscriber = nil
     }
 
-    /// Removes EngineWebView from MessageManager and sets the delegate to nil to stop receiving JS events.
-    func removeEngineWebView() {
-        // If delegate is nil, then EngineWebView is already cleaned up.
-        guard let _ = engine.delegate else { return }
+    // MARK: - Engine Cleanup
 
-        logger.logWithModuleTag("Cleaning EngineWebView from MessageManager", level: .debug)
+    open func removeEngineWebView() {
+        guard engine.delegate != nil else { return }
+        logger.logWithModuleTag("Cleaning EngineWebView from BaseMessageManager", level: .debug)
         engine.cleanEngineWeb()
         engine.delegate = nil
     }
 
-    func showMessage() {
-        elapsedTimer.start(title: "Displaying modal for message: \(currentMessage.messageId)")
+    // MARK: - Callbacks to be overridden by Subclasses
+
+    /// Called when the state changes to `.displayed`.
+    /// Subclasses decide how to show the UI (inline or modal).
+    open func onMessageDisplayed() {
+        // Subclasses override for their own show logic
+    }
+
+    // Public or open, no parameter:
+    open func onMessageDismissed() {
+        // Subclasses override
     }
 
-    private func loadModalMessage() {
-        guard isMessageLoaded else {
-            logger.logWithModuleTag("Message already loaded. Skipping loading modal message: \(currentMessage.describeForLogs)", level: .debug)
-            return
+    // Internal, for internal usage:
+    func onMessageDismissed(messageState: MessageState) {
```

Comment:
Can we please reconsider names of these methods? Current names are quite similar and make it difficult to differentiate between them. I'm fine with using longer, more descriptive names for internal methods if it improves clarity. Right now, `onMessageDismissed` and `onMessageDismissed` are hard to distinguish which can cause confusion when debugging or working on them.

Reply:
sure removed the unused one